src/components/common/TmSessionExtension.vue

```diff
@@ -76,13 +76,31 @@ export default {
       })
     },
     async signInAndRedirect(account) {
+      let accountNetwork = this.networks.find(
+        ({ id }) => id === account.network
+      )
+      // if no network, trying to identify by network_prefix
+      if (!accountNetwork) {
+        // select mainnet first
+        accountNetwork = this.networks.find(
+          ({ address_prefix, testnet }) =>
+            account.address.startsWith(address_prefix) && !testnet
+        )
+        if (!accountNetwork) {
+          // check for testnet
+          accountNetwork = this.networks.find(({ address_prefix }) =>
+            account.address.startsWith(address_prefix)
+          )
+        }
+      }
+      if (!account.network && accountNetwork) {
+        account.network = accountNetwork.id
+      }
       await this.signIn(account)
       this.$router.push({
         name: "portfolio",
         params: {
-          networkId:
-            this.networks.find(({ id }) => id === account.network).slug ||
-            "cosmos-hub" // defaulting to cosmos-hub-mainnet
+          networkId: accountNetwork ? accountNetwork.slug : "cosmos-hub" // defaulting to cosmos-hub-mainnet
```

Comment:
in which case is `accountNetwork` undefined?

Reply:
If someone will use a network account that was deleted

Reply:
so that should instead throw an error then "Network you are trying to sign into is not active on Lunie anymore".

Reply:
There is no error system in fe rn (

Reply:
but the error can be caught here in the component, right?

Reply:
Yap, but need to show something to user, cause now it's acting like broken link